src/DataFilterer.ts

```diff
@@ -83,6 +82,72 @@ export class DataFilterer {
         return [datasets, compVars, totals];
     }
 
+    calculateMean(filterOptions: DataFiltererOptions,
+                  impactData:    ImpactDataRow[],
+                  plotColours:   { [p: string] : string }): any[] {
+        // compare will always be year!
+
+        const filtData = this.filterByAll(filterOptions, impactData);
+
+        let meanVars = this.meanVariables(filterOptions.metric);
+        const top = meanVars.top;
+        const bottom = meanVars.bottom;
+        
+        const temp_top = this.filterByCompare(-1, filterOptions.compare, top, filtData);
+        const compVars_top: any[] = temp_top[1];
+        const filteredData_top: ImpactDataRow[] = temp_top[0];
+        // get an array of all the remaining disagg values
+        const aggVars_top: any[] = [...this.getUniqueVariables(-1, filterOptions.disagg, top, filteredData_top)];
+        const dataByAggregate_top = this.groupDataByDisaggAndThenCompare(filterOptions.compare, filterOptions.disagg, aggVars_top, filteredData_top);
+
+        const temp_bottom = this.filterByCompare(-1, filterOptions.compare, bottom, filtData);
```

Comment:
Looks like lines 103 - 109 contain the same logic as 96 - 101 with just a few variables changed - could they be factored out into a re-usable function? As well as reducing code duplication I think it would make this long function easier to understand!